src/fidesops/ops/models/privacy_request.py

```diff
@@ -505,27 +506,50 @@ def cache_manual_webhook_input(
             parsed_data.dict(),
         )
 
-    def get_manual_webhook_input(
+    def get_manual_webhook_input_strict(
         self, manual_webhook: AccessManualWebhook
     ) -> Dict[str, Any]:
-        """Retrieve manually added data that matches fields supplied in the specified manual webhook.
+        """
+        Retrieves manual webhook fields saved to the privacy request in strict mode.
+        Fails either if extra saved fields are detected (webhook definition had fields removed) or fields were not
+        explicitly set (webhook definition had fields added). This mode lets us know if webhooks data needs to be re-uploaded.
 
-        This is for use by the *manual_webhook* connector which is *NOT* integrated with the garph.
+        This is for use by the *manual_webhook* connector which is *NOT* integrated with the graph.
         """
-        cache: FidesopsRedis = get_cache()
-        cached_results: Optional[
-            Optional[Dict[str, Any]]
-        ] = cache.get_encoded_objects_by_prefix(
-            f"WEBHOOK_MANUAL_INPUT__{self.id}__{manual_webhook.id}"
+        cached_results: Optional[Dict[str, Any]] = _get_manual_input_from_cache(
+            privacy_request=self, manual_webhook=manual_webhook
         )
+
         if cached_results:
-            return manual_webhook.fields_schema.parse_obj(
-                list(cached_results.values())[0]
-            ).dict()
+            data: Dict[str, Any] = manual_webhook.fields_schema.parse_obj(
+                cached_results
+            ).dict(exclude_unset=True)
+            if set(data.keys()) != set(manual_webhook.fields_schema.__fields__.keys()):
+                raise ManualWebhookFieldsUnset(
+                    f"Fields unset for privacy_request_id '{self.id}' for connection config '{manual_webhook.connection_config.key}'"
+                )
+            return data
         raise NoCachedManualWebhookEntry(
             f"No data cached for privacy_request_id '{self.id}' for connection config '{manual_webhook.connection_config.key}'"
         )
 
+    def get_manual_webhook_input_non_strict(
```

Comment:
Added a "non strict" mode to load webhook data where we really just preserve the overlap.